app/queries/keyset_pagination/get_editions.rb

```diff
@@ -0,0 +1,86 @@
+module Queries
+  class KeysetPagination::GetEditions
+    attr_reader :fields
+
+    def initialize(params)
+      @fields = params[:fields] || default_fields
+      @order = params[:order] || "updated_at"
+      @filters = {
+        states: params[:states] || %i(draft published unpublished),
+        locale: params[:locale],
+        publishing_app: params[:publishing_app],
+      }
+
+      validate_fields!
+    end
+
+    def call
+      editions
+    end
+
+    def pagination_order
+      @pagination_order ||= (order.first == "-" ? :desc : :asc)
+    end
+
+    def pagination_key
+      @pagination_key ||= (
+        hash = {}
+        hash[pagination_field] = "editions.#{pagination_field}"
+        hash[:id] = "editions.id"
+        hash
+      )
+    end
+
+  private
+
+    attr_reader :order, :filters
+
+    DEFAULT_FIELDS = [
+      *Edition::TOP_LEVEL_FIELDS,
+      :content_id,
+      :locale,
+      :updated_at,
+      :created_at,
+    ].freeze
+
+    def pagination_field
+      @pagination_field ||= (if order.first == "-"
+                               order[1..order.length]
+                             else
+                               order
+                             end).to_sym
```

Comment:
Have added a restriction on the order by fields.

Not sure about using `begin ... end` as I don't think it'll make it any clearer or succinct, it'll look like this:

```
begin
  if order.first == "-"
    field = order[1..order.length]
  else
    field = order
  end
  field.to_sym
end
```